llvm/lib/Object/ArchiveWriter.cpp

```diff
@@ -678,6 +678,25 @@ static bool isECObject(object::SymbolicFile &Obj) {
   return false;
 }
 
+static bool useECMap(object::SymbolicFile &Obj) {
+  if (Obj.isCOFF())
+    return COFF::isAnyArm64(cast<COFFObjectFile>(&Obj)->getMachine());
```

Comment:
If I'm following correctly, this unconditionally emits an EC map for arm64 targets, even if there aren't any non-native object files?  But that isn't consistent with the tests, so I think I must be missing something.

Reply:
We don't actually emit EC symbol map if there are no EC symbols, so I wrote it under assumption that it should be fine. Thinking about it again, with #84834, this patch causes copied import descriptor to populate and emit EC map. I will fix it, thanks.